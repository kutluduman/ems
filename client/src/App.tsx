import "./App.css";
import Router from "./routes";
import ThemeConfig from "./theme";

const App = () => {
  return (
    <ThemeConfig>
      <Router />
    </ThemeConfig>
  );
}
export default App;