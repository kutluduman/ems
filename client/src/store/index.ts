import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import auth from "./slices/auth";
import employee from "./slices/employee";

export const store = configureStore({
  reducer: {
    auth,
    employee,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;