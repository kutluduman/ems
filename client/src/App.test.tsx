import { BrowserRouter as Router } from "react-router-dom";
import { create } from "react-test-renderer";
import { Provider } from "react-redux";

import { store } from "./store";
import App from "./App";

describe("App Component", () => {
  test("renders the App Component", () => {
    const tree = create(
      <Provider store={store}>
        <Router>
          <App />
        </Router>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});