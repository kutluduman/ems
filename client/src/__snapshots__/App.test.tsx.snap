// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders the App Component 1`] = `
<header
  className="css-z2vd3n"
>
  <a
    href="/"
    onClick={[Function]}
  />
</header>
`;
